cli/targets/static.js

```diff
@@ -690,13 +730,32 @@ function buildService(ref, service) {
             "@function " + lcName,
             "@memberof " + exportName(service),
             "@instance",
-            "@param {" + exportName(method.resolvedRequestType, !config.forceMessage) + "} request " + method.resolvedRequestType.name + " message or plain object",
-            "@returns {Promise<" + exportName(method.resolvedResponseType) + ">} Promise",
+            "@param {" +  exportTypeWithProtoFromIfNeeded(method.resolvedRequestType, !config.forceMessage) + "} request " + method.resolvedRequestType.name + " message or plain object",
+            "@returns {Promise<" + exportTypeWithProtoFromIfNeeded(method.resolvedResponseType) + ">} Promise",
             "@variation 2"
         ]);
     });
 }
 
+function exportTypeWithProtoFromIfNeeded(type, asInterface) {
+    let exportedType = exportName(type, asInterface);
+    if (type.undefinedInMainProto) {
+        exportedType = type.protoFrom + "." + exportedType;
+    }
+    return exportedType;
+}
+
+
+function exportResolvedType(type, asInterface) {
+    let exportedType = exportName(type, asInterface);
+    if (type.undefinedInMainProto) {
+        exportedType = type.protoFrom + "." + exportedType;
+    } else {
+        exportedType = "$root." + exportedType;
+    }
+    return exportedType;
```

Comment:
```suggestion
    if (type.undefinedInMainProto) {
        return type.protoFrom + "." + exportedType;
    }
    return "$root." + exportedType;    
```
or even better:
```suggestion
    return `${ type.undefinedInMainProto ? type.protoFrom : '$root'}.${exportedType}`;
```
Pay attention to the `.` sign between `${}` and `${}` above